addon/components/ember-wormhole.js

```diff
@@ -1,25 +1,90 @@
 import Ember from 'ember';
+import layout from '../templates/components/ember-wormhole';
 
 var computed = Ember.computed;
 var observer = Ember.observer;
 var run = Ember.run;
 
+function getActiveElement() {
+  if (typeof document === 'undefined') {
+    return null;
+  } else {
+    return document.activeElement;
+  }
+}
+
+// used to find an element by id
+function childNodesOfElement(element) {
+  let children = [];
+  let child = element.firstChild;
+  while (child) {
+    children.push(child);
+    child = child.nextSibling;
+  }
+  return children;
+}
+
+// implement our own getElementById since it doesn't exist in SimpleDOM
+function findElementById(doc, id) {
+  let nodes = childNodesOfElement(doc);
+  let node;
+
+  while (nodes.length) {
+    node = nodes.shift();
+
+    if (node.getAttribute && node.getAttribute('id') === id) {
+      return node;
+    }
+
+    nodes = childNodesOfElement(node).concat(nodes);
+  }
+}
+
 export default Ember.Component.extend({
+  layout,
   to: computed.alias('destinationElementId'),
   destinationElementId: null,
   destinationElement: computed('destinationElementId', 'renderInPlace', function() {
-    return this.get('renderInPlace') ? this.element : document.getElementById(this.get('destinationElementId'));
+    let renderInPlace = this.get('renderInPlace');
+    if (renderInPlace) {
+      return this._element;
+    }
+    let id = this.get('destinationElementId');
+    if (!id) {
+      return null;
+    }
+    let doc = this.get('dom').document;
+    let el = findElementById(doc, id);
+    return el;
   }),
   renderInPlace: false,
 
-  didInsertElement: function() {
+  init() {
     this._super(...arguments);
-    this._firstNode = this.element.firstChild;
-    this._lastNode = this.element.lastChild;
-    this.appendToDestination();
+    // keep a ref to 'dom' and 'registry' because those will be passed to
+    // the ember-wormhole-placeholder helper
+    let dom = this.renderer._dom;
+    this.set('dom', dom);
+    this.set('registry', {});
+    this._didInsert = false;
```

Comment:
instead of the registry do:
```
this._head = this._tail = undefined;
this._createHead = () => {
  return this._head = dom.createTextNode("");
};
this._createTail = () => {
   return this._tail = dom.createTextNode("");
};
```